tools/run_tests/dockerize/build_docker_and_run_tests.sh

```diff
@@ -33,7 +33,7 @@ DOCKER_IMAGE_NAME=$(basename "$DOCKERFILE_DIR")_$(sha1sum "$DOCKERFILE_DIR/Docke
 if [ "$DOCKERHUB_ORGANIZATION" != "" ]
 then
   DOCKER_IMAGE_NAME=$DOCKERHUB_ORGANIZATION/$DOCKER_IMAGE_NAME
-  time docker pull "$DOCKER_IMAGE_NAME"
+  time docker pull "$DOCKER_IMAGE_NAME" || time docker build -t "$DOCKER_IMAGE_NAME" "$DOCKERFILE_DIR"
```

Comment:
Can we adopt some mechanism here? If the script is run manually, we allow build the image on the fly. But if the script is executing on Kokoro, we have to upload the image.

Reply:
I've written a design doc about this. Because of credentials, this is trickier than you'd think.